### PR TITLE
Interface navigation implemented using recursion 🔨.

### PR DESCRIPTION
**Pull request by divyanshu-rawat**
Interface navigation recursion implementation 🔨

### CODE REVIEW

**Comment by Marvin9** [comment]
Pulled & tested locally. Working as expected. Overall Adding margin-left stuff to make it more clear. Overall perfect 👍

File: src/ignitus-UserInterfaceBook/InterfaceSideNavigation/constants.ts [hunk 1]
```diff
@@ -1,235 +1,195 @@
...
+      {
+        title: 'Separator',
+        route: '/interface/seperators',
+      },
+      {
+        title: 'Input Fields',
+        children: [
+          {
+            title: 'Search',
+            children: [{title: 'a', route: 'x'}, {title: 'c', route: 'z'}],
```
Comment (Marvin9) [inline src/ignitus-UserInterfaceBook/InterfaceSideNavigation/constants.ts:36]: Don't forget to remove this 😜

File: src/ignitus-UserInterfaceBook/InterfaceSideNavigation/constants.ts [hunk 2]
```diff
@@ -1,235 +1,195 @@
...
+        title: 'Navigation Drawer',
+        route: '/interface/checkboxes',
+      },
+      {
+        title: 'Footer',
+        route: '/interface/footer',
+      },
+      {
+        title: 'Card Layouts',
+        route: '',
```
Comment (Marvin9) [inline src/ignitus-UserInterfaceBook/InterfaceSideNavigation/constants.ts:135]: This too.

**Comment by Marvin9** [comment]
@divyanshu-rawat I will add margins once you merge to develop.

**Comment by divyanshu-rawat** [comment]
@Marvin9 merged then 🧨

**Pull request merged**